### PR TITLE
Fix negative zero decimals

### PR DESCRIPTION
**Pull request by simonz-bq**
- Fix failing tests for negative zero decimals in Ion Hash
- Remove not-supported target frameworks for test project
- Add tests for ToString of negative zero decimals
- Clean up some unreachable code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Amazon.IonDotnet.Tests/Amazon.IonDotnet.Tests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks Condition="'$(OS)' != 'Unix'">netcoreapp2.1;netcoreapp2.0;netcoreapp1.0;net45;net46</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' == 'Unix'">netcoreapp2.1;netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Unix'">netcoreapp2.1;net45;net46</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Unix'">netcoreapp2.1</TargetFrameworks>
```
Comment (tgregg) [inline Amazon.IonDotnet.Tests/Amazon.IonDotnet.Tests.csproj]: Did you intend for this change to be included in this PR?

Comment (simonz-bq) [reply Amazon.IonDotnet.Tests/Amazon.IonDotnet.Tests.csproj]: Yes. Tests do not compile due to .netstandard1.5 being removed from IonDotnet project in a recent PR. .netcoreapp1.0 was is not compatible with .netstandard2.0.

Removing .netcoreapp2.0 is just a "while we're in there" change, since it doesn't make sense to test using a framework that is out-of-support and superseded by .netcoreapp2.1.

Comment (tgregg) [reply Amazon.IonDotnet.Tests/Amazon.IonDotnet.Tests.csproj]: I don't have a problem with the change itself, but typically we'd want an unrelated change like this to be in its own PR (or, at the very least, in its own commit) to make the history more clear.
